### ISSUE TITLE
System audio becomes noisy on tdesktop launch

### ISSUE THREAD

**Issue by andreyorst** [telegramdesktop/tdesktop#3082]
That's only happen when I use my external audio card Scarlett 2i4 2nd Gen by Focusrite. And I use it for like 99% of my time. And every time i launch Telegram app my audio becomes noisy. This audio card is supported by Linux and don't require any drivers for it to work. Can't try to reproduce it in Windows, sorry, not a Windows user. Seems quite similar to #3081, but it's relatet to Windows so i let you decide is it similar issue, or not.

### Steps to reproduce
1. Plug in Scarlett 2i4 2nd Gen soundcard
2. Play some music or whatever that makes sound
3. Launch Telegram
4. That's it, now your audio screwed up, now unplug your card and plug back in

### Configuration
**Operating system:**
 Linux - Manjaro Testing (reproduces on stable too)
**Version of Telegram Desktop:**
1.0.14 binary version from site (Not AUR)

**Comment by mikhailnov** [comment]
Try to have a look at active audio output streams using pavucontrol when its noisy

**Comment by n1zzo** [comment]
I am experiencing the same issue.

**Operating system**:
Gentoo Linux (4.10.0) and Arch Linux (4.9.8-1)
**Version of Telegram Desktop**:
Telegram Desktop 1.0.14
**Additional Information**:
My audio card is a scarlett 2i2

**Comment by cmoro-deusto** [comment]
Count me in, but it doesn't reproduce 100% of the times I open Telegram when another app is using sound (ie Spotify, i used that to produce this bug report). 

**Operating system:**
Ubuntu Budgie 17.04
`Linux moria 4.10.0-20-generic #22-Ubuntu SMP Thu Apr 20 09:22:42 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux`
**Version of Telegram Desktop:**
Telegram Desktop 1.0.29
**Additional Information:**
My audio card is a PreSonus AudioBox USB (no drivers needed, supported directly in my distro).
Checked output streams in pavucontrol as suggested by @mikhailnov while the problem was reproducing and only the Spotify stream was shown, but please check second workaround below.
**I can workaround the issue in two ways**
   - if I close Telegram and the audio app and reopen them, as this doesnt happen everytime.
   - Launching pavucontrol (!) fixes the problem without closing neither Telegram nor Spotify

**Comment by andreyorst** [comment]
I can confirm that running pavucontrol fixes the problem without having to close tdesktop or turn off the usb sound card. But that's not the best way to fix such a problem anyway

**Comment by cmoro-deusto** [comment]
Yeah... @andreyorst I just was sharing that as a workaround and, primarily, to hopefully give some clues to devs so they can fix this! :)

**Comment by Matttter** [comment]
I have this 100% of the time, regardless of where my audio comes from. I use GalliumOS (ubuntu 16.04). It's very annoying

**Comment by andreyorst** [comment]
@Matttter can you please try to reproduce theese bugs #3458 #3459?

**Comment by jgreitemann** [comment]
I have a similar problem. Audio cuts out on launch of Telegram (cf. #3081 ) and appears *higher pitched* afterwards. It doesn't seem noisy just a lot higher. Not sure if this is related but this thread tipped me off towards opening `pavucontrol` and this indeed fixes my problem as well.

**Comment by ntonnaett** [comment]
Same here. Higher pitch when opening telegram, Fedora 27, Focusrite Scarlett 2i4 1st Gen, starting pavucontrol fixes the pitch. But I really don't know if this is the same bug. That looks more like a problem with setting the right sampling rate.

**Comment by andreyorst** [comment]
Actually I've encountered this problem when launching qTox. Maybe It is not Telegram related. If anyone can check it, feedback would be much appreciated.

**Comment by mtaalas** [comment]
Telegram also causes completely unnecessary sample rate changes on windows platform when it starts... maybe I should add another issue for windows only?

I use MOTU 828MKII for music production and I want to run everything in one single and known sample rate that doesn't change on it's own (48/96khz). Telegram always resets this to 44.1khz when starting no matter what I do in any other sound setting available to me. And this makes telegram a very bad choice to be run in my main computer since it can do this on it's own suddenly and to reset the sampling rate i have to close all programs using the audio device first.

Telegram should always respect the sampling rate of the system before trying to change it at all. If it can't respect that, there has to be sample rate conversion added to the code such that it can handle running in a system with different sampling rates even if it's using 44.1khz internally.

It's completely inexcusable as it stands. :/

**Comment by andreyorst** [comment]
@mtaalas Indeed. And pitch issues, that were reported by @jgreitemann and @ntonnaett more likely related to sampling rate, because changing it may change the pitch if software doesn't notified about this change. I've wrote a VST plugin, where I were calculating the frequency by counting samples, and knowing a sample size allowed me to calculate frequency (dirty hack, but I was lazy to mess with FFT). And changing sample rate resulted in a wrong calculated frequency, and the pitch was wrong.
